### ISSUE TITLE
bdist_conda pins numpy unnecessarily

### ISSUE THREAD

**Issue by jakirkham** [conda/conda-build#3333]
### Actual Behavior

Appears that `bdist_conda` will pin `numpy` in the package if it shows up in `install_requires` even if it was not otherwise constrained by the user.

### Expected Behavior

`numpy` should not get pinned.

### Steps to Reproduce

Used the `setup.py` file below and running `python setup.py bdist_conda` in the `base` environment. The resulting package has `numpy 1.11.*` in the requirements even though that was not specified by the user. (Was using the `condaforge/linux-anvil` image for reproducibility.)

```python
from setuptools import setup

from distutils.command.bdist_conda import CondaDistribution as Distribution


setup(
    name="test_setup_pkg",
    version="1.0.0",
    build_requires=[],
    install_requires=["numpy"]
)
```

##### `Output of conda info`

```
$ conda info

     active environment : base
    active env location : /opt/conda
            shell level : 1
       user config file : /home/conda/.condarc
 populated config files : /home/conda/.condarc
          conda version : 4.5.11
    conda-build version : 3.17.5
         python version : 3.6.7.final.0
       base environment : /opt/conda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /opt/conda/pkgs
                          /home/conda/.conda/pkgs
       envs directories : /opt/conda/envs
                          /home/conda/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.11 requests/2.21.0 CPython/3.6.7 Linux/4.9.93-linuxkit-aufs centos/6.10 glibc/2.12
                UID:GID : 9001:9001
             netrc file : None
           offline mode : False
```

**Comment by jakirkham** [comment]
The following change to the `setup.py` above seems to workaround the issue.

```diff
10c10
<     install_requires=["numpy"]
---
>     install_requires=["numpy >=1.11"]
```

**Comment by gregmuellegger** [comment]
Hi, I stumbled upon the very same problem. And as @jakirkham writes, the problem is resolved when using `install_requires=["numpy >=1.11"]`.

However the message when running `bdist_conda` regarding pinning numpy remains: `No numpy version specified in conda_build_config.yaml. Falling back to default numpy value of 1.11`

Now the problem I have is that I'm maintaining quite a few packages and when I forget to specify the "numpy>=1.11", I run into troubles that all my package users get downgraded to numpy 1.11.

Is there any particular reason for pinning numpy so hard in conda? How can disable that without touch every single `setup.py` file?

**Comment by bilderbuchi** [comment]
Funny thing, I even get these messages when having `numpy>1.11` (not `>=`) in `install_requires` (using conda-build 3.18.10).
